### PR TITLE
Remove unused import detected by nightly rust

### LINKED ISSUE #5476: Unused import in nightly rust
**Describe the bug**

With this pr been merged to nightly rust: https://github.com/rust-lang/rust/pull/117772
The tracking for unused import has been improved, resulting numerous compiler warnings when building arrow-rs with nightly rust.

**To Reproduce**

```rust
rustup default nightly
cargo check
```

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by XiangpengHao**
# Which issue does this PR close?

Closes #5476.

# Rationale for this change

With this pr been merged to nightly rust: https://github.com/rust-lang/rust/pull/117772
The tracking for unused import has been improved, resulting numerous compiler warnings when building arrow-rs with nightly rust.

 This PR simply remove them.

Note that there're many other linting problems as well in nightly rust, this pr only fix the problem with unused import, considering that the changes are quite large already.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by XiangpengHao** [comment]
hmm, the CI failed with errors like 

error: use of deprecated associated function `chrono::TimeDelta::milliseconds`: Use `TimeDelta::try_milliseconds` instead

It does not seem to relate to the changes

**Comment by XiangpengHao** [comment]
Ok the latest CI uses chrono v0.4.35 while previous ci uses v0.4.34.

It turns out that they release the new version a few hours ago and deprecated a few functions used by arrow-rs: https://github.com/chronotope/chrono/releases/tag/v0.4.35

**Comment by tustvold** [comment]
Should be fixed if you merge master

**Comment by XiangpengHao** [comment]
Can @tustvold  you take a look at `make_fixed_len_byte_array_reader` in the [bench file](https://github.com/apache/arrow-rs/pull/5477/files#diff-6e324bad65a3e04c05874112fdef376b6eea7f5f6ad7c0c21843b54b41675c68R25)? The function is from [array_reader](https://github.com/apache/arrow-rs/pull/5477/files#diff-80684f99294e6b631ba411e76ee05998a7af0b49914cee1ef6cc930300d4e38aR51) and is only used in the benchmark. I can't just remove the export as it will fail the benchmark, so I just suppressed the warning. Not sure it is the right way to do...

File: parquet/src/compression.rs [hunk 1]
```diff
@@ -441,8 +437,6 @@ mod lz4_codec {
         }
     }
 }
-#[cfg(any(feature = "lz4", test))]
-pub use lz4_codec::*;
```
[the remark this answers is not in the document]
Comment (XiangpengHao) [reply parquet/src/compression.rs:445]: I tried with lz4 on and off, it seems that both are not used.

[the remark this answers is not in the document]
Comment (tustvold) [reply parquet/src/compression.rs:445]: But this is a public export when the experimental flag is enabled

[the remark this answers is not in the document]
Comment (tustvold) [reply parquet/src/compression.rs:445]: It is public when the experimental feature is enabled

https://github.com/apache/arrow-rs/blob/master/parquet/src/lib.rs#L124

[the remark this answers is not in the document]
Comment (XiangpengHao) [reply parquet/src/compression.rs:445]: Ah you are right! I didn't see two experimental macro there...

**Comment by tustvold** [comment]
Thank you

**Pull request merged**